resources/qml/Menus/ConfigurationMenu/ConfigurationItem.qml

```diff
@@ -40,19 +57,99 @@ Button
                 right: parent.right
                 rightMargin: UM.Theme.getSize("wide_margin").width
             }
-
+            height: childrenRect.height
             spacing: UM.Theme.getSize("default_margin").width
 
             Repeater
             {
                 id: repeater
-                model: configuration.extruderConfigurations
+                model:
+                {
+                    if (configurationItem.isValidMaterial)
+                    {
+                        return configuration.extruderConfigurations
+                    }
+                    return []
```

Comment:
Why change the model instead of changing the delegate? 

It's not that there is no configuration, the point is that we need to show a different call to action if the material is not valid.

Reply:
Fixed: 5e9854454185fee9490b0d9aa996d36f65b7e2fd